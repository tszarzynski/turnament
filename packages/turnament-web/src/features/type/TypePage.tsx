import { Button, Hr, IconNext } from "turnament-components";
import type { SchedulerType } from "turnament-scheduler";
import { routes } from "../../app/router";
import { useBaseStore } from "../../app/store";
import PageHeader from "../../components/PageHeader";
import { schedulerOptions } from "../../utils/schedulerUtils";
import PageLayout, { PageContent, PageBody } from "../../components/PageLayout";
import PageNavigation from "../../components/PageNavigation";

const TypePage = () => {
	const nextRound = useBaseStore((state) => state.nextRound);
	const schedulerType = useBaseStore((state) => state.schedulerType);
	const setSchedulerType = useBaseStore((state) => state.setSchedulerType);

	const handleOptionChange = (event: React.ChangeEvent<HTMLInputElement>) => {
		setSchedulerType(event.target.value as SchedulerType);
	};

	const handleNext = () => {
		nextRound();
		routes.tournamentRounds().push();
	};

	return (
		<PageLayout>
			<PageContent>
				<PageHeader>Choose Type</PageHeader>
				<PageBody>
					<fieldset>
						<div className="border-2 border-secondary p-0.5">
							<Hr />
							<div className="p-0.5">
								<legend className="flex w-[calc(100%-56px)] items-center border border-secondary px-4 py-2 text-center font-bold text-secondary text-upright uppercase [letter-spacing:-4px]">
									Type
								</legend>
								{schedulerOptions.map((option) => (
									<div
										key={option.value}
										className="mt-0.5 flex w-full flex-row items-stretch justify-between "
									>
										<label
											htmlFor={option.value}
											className="flex flex-1 items-center border border-secondary px-4 py-3 font-medium text-lg text-secondary uppercase"
										>
											{option.name}
										</label>
										<input
											id={option.value}
											name="tournament-type"
											type="radio"
											value={option.value}
											checked={schedulerType === option.value}
											onChange={handleOptionChange}
											className="h-[42px] w-[42px] mr-3 ml-0.5 bg-primary accent-black focus:outline-none focus:ring-2 focus:ring-black focus:ring-offset-1"
										/>
									</div>
								))}
							</div>
						</div>
					</fieldset>
				</PageBody>
			</PageContent>
			<PageNavigation>
				<Button
					type="button"
					fullWidth={false}
					iconSlot={<IconNext />}
					onClick={handleNext}
				>
					Start
				</Button>
			</PageNavigation>
		</PageLayout>
	);
};

export default TypePage;
