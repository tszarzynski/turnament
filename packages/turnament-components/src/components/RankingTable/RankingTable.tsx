import type { PlayerWithStats } from "turnament-scheduler";
import Hr from "../Hr";

type Props = {
	playersWithStats: PlayerWithStats[];
};

const columns = ["Rank", "Name", "Wins", "Pts", "OMV"];

const RankingTable = ({ playersWithStats }: Props) => {
	return (
		<div className="w-full border-2 border-secondary p-0.5">
			<Hr />
			<table className="min-w-full table-auto border-separate border-spacing-0.5">
				<TableHead columns={columns} />
				<tbody>
					{playersWithStats.map((player, index) => (
						<tr key={player.ID} className="h-[54px] align-bottom">
							<td className="align w-[54px] border border-secondary px-4 py-1 text-center text-secondary ">
								{index + 1}
							</td>
							<td className="border border-secondary px-4 py-1 font-bold text-2xl text-handwritten leading-none">
								{player.name}
							</td>
							<td className="border border-secondary px-4 py-1 text-handwritten">
								{player.matchesWon}
							</td>
							<td className="border border-secondary px-4 py-1 text-handwritten">
								{player.gamesWon}
							</td>
							<td className="border border-secondary px-4 py-1 text-handwritten">
								{player.omv.toFixed(2)}
							</td>
						</tr>
					))}
				</tbody>
			</table>
		</div>
	);
};

export default RankingTable;

const TableHead = ({ columns }: { columns: string[] }) => {
	return (
		<thead>
			<tr>
				{columns.map((column) => (
					<th
						key={column}
						className={
							"border-2 border-secondary px-2 py-1 font-bold text-secondary text-upright text-xs"
						}
					>
						{column}
					</th>
				))}
			</tr>
		</thead>
	);
};
