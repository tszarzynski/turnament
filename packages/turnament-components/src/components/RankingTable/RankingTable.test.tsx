import { render } from "vitest-browser-react";
import type { PlayerWithStats } from "turnament-scheduler";
import RankingTable from "./RankingTable";

const playersWithStats: PlayerWithStats[] = [
	{
		ID: 1,
		name: "John Doe",
		active: true,
		gamesWon: 5,
		matchesLost: 6,
		matchesWon: 7,
		omv: 12.34,
		opponents: [],
	},
	{
		ID: 2,
		name: "Jane Smith",
		active: true,
		gamesWon: 8,
		matchesLost: 3,
		matchesWon: 9,
		omv: 15.21,
		opponents: [],
	},
	{
		ID: 3,
		name: "Mike Johnson",
		active: false,
		gamesWon: 4,
		matchesLost: 8,
		matchesWon: 4,
		omv: 10.76,
		opponents: [],
	},
	{
		ID: 4,
		name: "Emily Williams",
		active: true,
		gamesWon: 10,
		matchesLost: 2,
		matchesWon: 11,
		omv: 17.45,
		opponents: [],
	},
	{
		ID: 5,
		name: "Alex Brown",
		active: true,
		gamesWon: 7,
		matchesLost: 5,
		matchesWon: 6,
		omv: 13.89,
		opponents: [],
	},
	{
		ID: 6,
		name: "Sarah Davis",
		active: false,
		gamesWon: 3,
		matchesLost: 9,
		matchesWon: 3,
		omv: 9.52,
		opponents: [],
	},
];

describe("RankingTable", () => {
	it("renders without error", () => {
		const { asFragment } = render(
			<RankingTable playersWithStats={playersWithStats} />,
		);
		expect(asFragment()).toMatchSnapshot();
	});
});
