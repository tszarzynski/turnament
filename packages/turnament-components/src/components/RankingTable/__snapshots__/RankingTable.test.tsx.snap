// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`RankingTable > renders without error 1`] = `
<DocumentFragment>
  <div
    class="w-full border-2 border-secondary p-0.5"
  >
    <hr
      class="mb-0.5 border-secondary"
    />
    <table
      class="min-w-full border-spacing-0.5 border-separate table-auto"
    >
      <thead
        class="border border-secondary"
      >
        <tr>
          <th
            class="text-upright text-secondary text-xs py-2 px-4 font-bold uppercase border border-secondary"
          >
            Rank
          </th>
          <th
            class="text-upright text-secondary text-xs py-2 px-4 font-bold uppercase border border-secondary"
          >
            Name
          </th>
          <th
            class="text-upright text-secondary text-xs py-2 px-4 font-bold uppercase border border-secondary"
          >
            Wins
          </th>
          <th
            class="text-upright text-secondary text-xs py-2 px-4 font-bold uppercase border border-secondary"
          >
            Pts
          </th>
          <th
            class="text-upright text-secondary text-xs py-2 px-4 font-bold uppercase border border-secondary"
          >
            OMV
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td
            class="py-3 px-4 border border-secondary text-secondary text-center"
          >
            1
          </td>
          <td
            class="py-3 px-4 border border-secondary font-bold"
          >
            John Doe
          </td>
          <td
            class="py-3 px-4 border border-secondary"
          >
            7
          </td>
          <td
            class="py-3 px-4 border border-secondary"
          >
            5
          </td>
          <td
            class="py-3 px-4 border border-secondary"
          >
            12.34
          </td>
        </tr>
        <tr>
          <td
            class="py-3 px-4 border border-secondary text-secondary text-center"
          >
            2
          </td>
          <td
            class="py-3 px-4 border border-secondary font-bold"
          >
            Jane Smith
          </td>
          <td
            class="py-3 px-4 border border-secondary"
          >
            9
          </td>
          <td
            class="py-3 px-4 border border-secondary"
          >
            8
          </td>
          <td
            class="py-3 px-4 border border-secondary"
          >
            15.21
          </td>
        </tr>
        <tr>
          <td
            class="py-3 px-4 border border-secondary text-secondary text-center"
          >
            3
          </td>
          <td
            class="py-3 px-4 border border-secondary font-bold"
          >
            Mike Johnson
          </td>
          <td
            class="py-3 px-4 border border-secondary"
          >
            4
          </td>
          <td
            class="py-3 px-4 border border-secondary"
          >
            4
          </td>
          <td
            class="py-3 px-4 border border-secondary"
          >
            10.76
          </td>
        </tr>
        <tr>
          <td
            class="py-3 px-4 border border-secondary text-secondary text-center"
          >
            4
          </td>
          <td
            class="py-3 px-4 border border-secondary font-bold"
          >
            Emily Williams
          </td>
          <td
            class="py-3 px-4 border border-secondary"
          >
            11
          </td>
          <td
            class="py-3 px-4 border border-secondary"
          >
            10
          </td>
          <td
            class="py-3 px-4 border border-secondary"
          >
            17.45
          </td>
        </tr>
        <tr>
          <td
            class="py-3 px-4 border border-secondary text-secondary text-center"
          >
            5
          </td>
          <td
            class="py-3 px-4 border border-secondary font-bold"
          >
            Alex Brown
          </td>
          <td
            class="py-3 px-4 border border-secondary"
          >
            6
          </td>
          <td
            class="py-3 px-4 border border-secondary"
          >
            7
          </td>
          <td
            class="py-3 px-4 border border-secondary"
          >
            13.89
          </td>
        </tr>
        <tr>
          <td
            class="py-3 px-4 border border-secondary text-secondary text-center"
          >
            6
          </td>
          <td
            class="py-3 px-4 border border-secondary font-bold"
          >
            Sarah Davis
          </td>
          <td
            class="py-3 px-4 border border-secondary"
          >
            3
          </td>
          <td
            class="py-3 px-4 border border-secondary"
          >
            3
          </td>
          <td
            class="py-3 px-4 border border-secondary"
          >
            9.52
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
